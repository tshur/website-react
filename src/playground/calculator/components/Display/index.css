:local .display {
  background: #eee;
  height: 60px;
  padding: 0 5px;
  margin-bottom: 15px;
  border: 1px solid black;
}

:local .value {
  font-family: 'Courier New', Courier, monospace;
  font-size: 48px;
  text-align: right;
  line-height: 1.5;
}