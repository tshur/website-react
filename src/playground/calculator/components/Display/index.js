import React from 'react';
import styles from './index.css';

const Display = ({ value }) => {
  if (!isFinite(value))
    value = 'Error';

  return (
    <div className={styles.display}>
      <div className={styles.value}>{value}</div>
    </div>
  );
}

export default Display;
