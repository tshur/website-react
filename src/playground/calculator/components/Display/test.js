import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import Display from '.';

describe('Display', () => {

  test('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Display />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  test('has a valid snapshot', () => {
    const wrapper = shallow(
      <Display />
    );
    expect(wrapper).toMatchSnapshot();
  });

  test('represents numbers with many digits', () => {
    const wrapper = shallow(
      <Display />
    );
    const displayText = () => wrapper.find('div>div').text();

    // Large numbers
    wrapper.setProps({ value: 999999999 });
    expect(displayText()).toEqual('999999999');
    wrapper.setProps({ value: 1000000000 });
    expect(displayText()).toEqual('1.0000e+9');

    // Small numbers
    wrapper.setProps({ value: 0.1234567 });
    expect(displayText()).toEqual('0.1234567');
    wrapper.setProps({ value: 6.1234567 });
    expect(displayText()).toEqual('6.1234567');
    wrapper.setProps({ value: 0.0123456789 });
    expect(displayText()).toEqual('0.01234568');

    // Mixed precision
    wrapper.setProps({ value: 12345.6789 });
    expect(displayText()).toEqual('12345.6789');
    wrapper.setProps({ value: 12.3456789 });
    expect(displayText()).toEqual('12.3456789');

    // Negative numbers
    wrapper.setProps({ value: -123456789 });
    expect(displayText()).toEqual('-1.2346e+8');
  });
});
