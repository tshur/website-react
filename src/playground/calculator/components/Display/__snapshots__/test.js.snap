// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Display has a valid snapshot 1`] = `
<div>
  <div>
    Error
  </div>
</div>
`;
